modules/apps/feature-flag/feature-flag-web/src/main/java/com/liferay/feature/flag/web/internal/model/listener/CompanyModelListener.java

```diff
@@ -29,29 +36,60 @@ public class CompanyModelListener extends BaseModelListener<Company> {
 
 	@Override
 	public void onAfterCreate(Company company) throws ModelListenerException {
-		TransactionCommitCallbackUtil.registerCallback(
```

Comment:
@thiago-buarqque The original logic uses a `TransactionCommitCallback`; but you removed it in the change. Any reason?

If we see the company creation as a transaction, this will be called at the point [right after the new company model is added to the database](https://github.com/liferay/liferay-portal/blob/38220df7f113ab58af84b3297d81db6bb40d3320/portal-kernel/src/com/liferay/portal/kernel/service/persistence/impl/BasePersistenceImpl.java#L792). However, in this transaction, there might be many other creations/updates on the company or other entities after this. Removing the `TransactionCommitCallbackUtil` usage here will move the execution of this logic from the end of the big transaction (when it's committed) to the middle of the transaction (directly here). This would be a big change in the execution point of time.

If there's no good reason for it, please keep the `TransactionCommitCallbackUtil` usage in the `onAfterCreate`.

Reply:
No reason tbh, it was a mistake. Fixed it